compiler/env/mem/OMRBackingMemoryAllocator.cpp

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+
+#include <stdlib.h>
+#include "codegen/FrontEnd.hpp"                  // for feGetEnv
+
+// Memory logging support
+// Memory allocation can be used very early by the compiler: can rely on printf working
+
+static char *LOGLEVEL = ::feGetEnv("OMRDebug_BackingMemory");
+
+// LOGLEVEL="0"    // no debug output
+// LOGLEVEL="1"    // summary output, about one line per function call
+// LOGLEVEL="2"    // extra detail in functions
+// LOGLEVEL="3"    // highest level of detail in functions
+
+#define MEMLOG(n,s, ...) do { if ((LOGLEVEL != NULL) && ((int)(*LOGLEVEL) >= (int)(n+'0'))) { printf("BckMem %p : " s, this, __VA_ARGS__); } } while (0)
+#define MEMLOG0(n,s)     do { if ((LOGLEVEL != NULL) && ((int)(*LOGLEVEL) >= (int)(n+'0'))) { printf("BckMem %p : " s, this); } } while (0)
+
+
+#include "env/mem/BackingMemoryAllocator.hpp"
+
+
+TR::MemorySegment &
+OMR::BackingMemoryAllocator::allocate(size_t requiredSize, void * hint)
```

Comment:
This method doesn't check the `_allocationLimit` even though that concept exists at the `BackingMemoryAllocator` level.

Reply:
Um, yeah...good point :)